src/Agent.Worker/Build/BuildCommandExtension.cs

```diff
@@ -96,7 +94,7 @@ private void ProcessBuildUpdateBuildNumberCommand(IExecutionContext context, str
 
                 // queue async command task to update build number.
                 context.Debug($"Update build number for build: {buildId.Value} to: {data} at backend.");
-                var commandContext = HostContext.CreateService<IAsyncCommandContext>();
+                var commandContext = context.GetHostContext().CreateService<IAsyncCommandContext>();
```

Comment:
Do we need this change/what is it doing? Its unrelated to the conversion, right?

Reply:
it is part of the conversion. Since the command is now it's own class and not an IAgentService, it does not have a HostContext property. The HostContext is already part of the IExecutionContext object so we added a method to expose that and just use that rather than make each command an IAgentService.

Reply:
Ah, right - my bad